### PR TITLE
Add params to DeleteSnapshot API

### PR DESCRIPTION
**Pull request by deepakkinni**
1. Require params to deliver the backup repository.
2. No funtional change yet, only API defn changed.

Testing:
build.

Signed-off-by: Deepak Kinni <dkinni@vmware.com>

### CODE REVIEW

File: pkg/s3repository/repository_protected_entity.go [hunk 1]
```diff
@@ -602,7 +604,7 @@ func (this *ProtectedEntity) cleanupOnAbortedUpload(ctx *context.Context) {
 		log.Infof("The context was canceled during copy of pe %v, proceeding with cleanup", peInfo.GetName())
 		log.Debugf("Attempting to delete any uploaded snapshots for %v", this.peinfo.GetID())
 		// New context or else downstream "withContext" calls will error out.
-		status, err := this.DeleteSnapshot(context.Background(), this.peinfo.GetID().GetSnapshotID())
+		status, err := this.DeleteSnapshot(context.Background(), this.peinfo.GetID().GetSnapshotID(), nil)
```
Comment (dsu-igeek) [inline pkg/s3repository/repository_protected_entity.go:607]: We're passing in an empty map everywhere else.  Let's pass in the empty map here as well

File: pkg/ivd/ivd_protected_entity_test.go [hunk 2]
```diff
@@ -711,7 +711,7 @@ func TestBackupEncryptedIVD(t *testing.T) {
 	defer func() {
 		for snapPEID, _ := range snapPEIDtoIvdPEMap {
 			s3PE := snapPEIDtos3PEMap[snapPEID]
-			_, err := s3PE.DeleteSnapshot(ctx, snapPEID.GetSnapshotID())
+			_, err := s3PE.DeleteSnapshot(ctx, snapPEID.GetSnapshotID(), nil)
```
Comment (dsu-igeek) [inline pkg/ivd/ivd_protected_entity_test.go:714]: Let's pass in the empty map here as well

File: pkg/ivd/ivd_protected_entity_test.go [hunk 3]
```diff
@@ -720,7 +720,7 @@ func TestBackupEncryptedIVD(t *testing.T) {
 
 	// #5.3: Delete the local IVD snapshot
 	for snapPEID, ivdPE := range snapPEIDtoIvdPEMap {
-		_, err := ivdPE.DeleteSnapshot(ctx, snapPEID.GetSnapshotID())
+		_, err := ivdPE.DeleteSnapshot(ctx, snapPEID.GetSnapshotID(), nil)
```
Comment (dsu-igeek) [inline pkg/ivd/ivd_protected_entity_test.go:723]: Let's pass in the empty map here as well

File: pkg/s3repository/repository_protected_entity.go (hunk 1 above)
Comment (deepakkinni) [reply pkg/s3repository/repository_protected_entity.go]: Done. Here and elsewhere.

**Pull request merged**